src/components/CreateTemplate/index.tsx

```diff
@@ -0,0 +1,217 @@
+  const { userId, firstName, lastName } = useStoreSelector((state) => state.user);
+  const { activeCourt: selectedCourt } = useStoreSelector((state) => state.courtSpecData);
+  const { court: selectedCourtTileData } = useStoreSelector((state) => state.tile.present);
+  const [addTemplate] = useAddTemplateMutation();
+  const courtType = "basketball";
+  const userFullName = userNameEllip(`${firstName} ${lastName}`, maxUserNameDisplay);
+  const {
+    isOpen: isSuccessNoticeOpen,
+    onOpen: handleSuccessNoticeOpen,
+    onClose: handleSuccessNoticeClose,
+  } = useDisclosure();
+
+  const closeWindow = () => {
+    setInputError(inputErrorInit);
+    setTextAreaLen(0);
+    onClose();
+  };
+
+  const checkNameLength = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const nameInputLen = e.currentTarget.value.length;
+    setInputError((inputError) => ({
+      ...inputError,
+      courtNameFullErr: nameInputLen > maxCourtNameLen,
+      courtNameNullErr: nameInputLen === 0,
+    }));
+  };
+
+  const handleTextAreaLenChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    const textAreaLength = e.currentTarget.value
+      .replace(regex, " ")
+      .split(" ")
+      .filter((item) => item != "").length;
+    setTextAreaLen(textAreaLength);
+    setInputError((inputError) => ({
+      ...inputError,
+      descriptionOverLimit: textAreaLength > maxDescriptionLen,
+    }));
+  };
+
+  const submitTemplate = async () => {
+    const courtName = courtNameRef.current?.value;
+    const description = descriptionRef.current?.value;
+    if (!courtName || inputError.descriptionOverLimit || inputError.courtNameFullErr) {
+      setInputError((inputError) => ({
+        ...inputError,
+        courtNameNullErr: !courtName,
+      }));
+      return;
+    }
+    const packedTemplate = generateNewTemplate(
+      userId,
+      courtName,
+      description,
+      selectedCourtTileData,
+      selectedCourt
+    );
+    await addTemplate(packedTemplate)
+      .unwrap()
+      .then((_res) => {
+        handleSuccessNoticeOpen();
+      })
+      .catch((err) => {
+        alert("Whoops! unsuccessful publish your template, please try again!");
+      });
+    closeWindow();
+  };
+
+  return (
+    <>
+      <Modal isOpen={isOpen} onClose={closeWindow} size="2xl" isCentered>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader role="modalTitle">Template sharing</ModalHeader>
+          <ModalCloseButton />
+          <ModalBody padding="8px 24px">
+            <FormControl>
+              <FormLabel>Court Preview:</FormLabel>
+              <Box width={300} height={150} backgroundColor="orange" margin="30px auto">
+                Court Image
+              </Box>
+            </FormControl>
+            <Flex gap="24px" justifyContent="center" flexWrap="wrap">
+              <Badge margin="16px" colorScheme="green">
+                {courtType}
+              </Badge>
```

Comment:
const courtType = "basketball";
Why not just put this basketball here?

Reply:
Considering the template tags for courtType currently only have the basketball type(tennis or mixed court is added in the future). We might need to read the court type from the design specifications in the future. This const for courtType is just a temporary solution for the court type tag.